File: android/src/main/java/org/ligi/satoshiproof/FileProofController.java
package org.ligi.satoshiproof;

import android.app.Activity;
import android.support.v7.app.AlertDialog;
import java.io.File;
import org.apache.commons.io.FileUtils;
import org.ligi.axt.listeners.DialogDiscardingOnClickListener;

public class FileProofController {
    private final Activity context;

    public FileProofController(final Activity context) {
        this.context = context;
    }


    public void proofFile(final File file) {
        context.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                try {
                    if (file == null) {
                        failWitAlertDialog("Could not open file");
                        return;
                    }
                    final byte[] imageBytes = FileUtils.readFileToByteArray(file);
                    new ProofAsyncTask(context, imageBytes).execute();
                } catch (Exception e) {
                    failWitAlertDialog("Could not open " + file + " " + e);
                }
            }
        });
    }


    private void failWitAlertDialog(String msg) {
        new AlertDialog.Builder(context).setMessage(msg).setPositiveButton(android.R.string.ok, new DialogDiscardingOnClickListener()).show();
    }

}

File: android/src/main/kotlin/org/ligi/satoshiproof/FileProofController.kt
package org.ligi.satoshiproof

import android.app.Activity
import android.support.v7.app.AlertDialog
import org.apache.commons.io.FileUtils
import java.io.File

class FileProofController(private val context: Activity) {


    fun proofFile(file: File?) {
        context.runOnUiThread(Runnable {
            try {
                if (file == null) {
                    failWitAlertDialog("Could not open file")
                    return@Runnable
                }
                val imageBytes = FileUtils.readFileToByteArray(file)
                ProofAsyncTask(context, imageBytes).execute()
            } catch (e: Exception) {
                failWitAlertDialog("Could not open $file $e")
            }
        })
    }

    private fun failWitAlertDialog(msg: String) {
        AlertDialog.Builder(context).setMessage(msg).setPositiveButton(android.R.string.ok, null).show()
    }

}
